isis/src/lro/apps/lrowaccal/lrowaccal.cpp

```diff
@@ -0,0 +1,788 @@
+#include "lrowaccal.h"
+
+#define POLAR_MODE_SAMPLES 1024
+#define NO_POLAR_MODE_SAMPLES 704
+#define BW_BANDS 1
+#define VIS_LINES 14
+#define COLOR_BANDS 5
+#define UV_SAMPLES 128
+#define UV_LINES 4
+#define UV_BANDS 2
+#define KM_PER_AU 149597871
+
+using namespace std;
+using namespace Isis;
+
+
+namespace Isis {
+  // globals
+
+  /**
+   * @brief Calibrate a WAC cube. 
+   *
+   * This is the programmatic interface to the lrowaccal application.
+   * 
+   * @param ui the User Interface to parse the parameters from
+   */
+  void lrowaccal(UserInterface &ui) {
+    Cube *icube = NULL;
+    icube = new Cube();
+    icube->open(ui.GetCubeName("FROM"));
```

Comment:
You need to get the input attributes too. You can copy how cam2map does it https://github.com/USGS-Astrogeology/ISIS3/blob/dev/isis/src/base/apps/cam2map/cam2map.cpp#L25

Reply:
lrowaccal gets the input attributes now like how cam2map does it